tests/test_flows.py

```diff
@@ -220,30 +220,35 @@ def init(self):
         self.mac_repeater2_wlan2 = re.search(rb"link/ether " + RE_MAC, mac_repeater2_wlan2_output).group('mac').decode()
         debug("Repeater2 wl2: {}".format(self.mac_repeater2_wlan2))
 
-    def _check_log_internal(self, device: str, program: str, regex: str) -> bool:
+    def _check_log_internal(self, device: str, program: str, regex: str, start_line: int):
         '''Search for regex in logfile for program on device.'''
         logfilename = os.path.join(self.rootdir, 'logs', device, 'beerocks_{}.log'.format(program))
         with open(logfilename) as logfile:
-            for line in logfile.readlines():
-                if re.search(regex, line):
+            for (i, v) in enumerate(logfile.readlines()):
+                if i <= start_line:
+                    continue
+                search = re.search(regex, v)
+                if search:
                     debug("Found '{}'\n\tin {}".format(regex, logfilename))
-                    return True
-        return False
+                    return (True, i, search.groups())
+        return (False, start_line, None)
```

Comment:
In general, you don't know if there will be groups in the match object. For full flexibility, just return the match object. That also removes the need to have a True, False as first return value, because the None already functions as a truth value.

Also, I think it makes much more sense to return the last line in the failure case, because that allows us to continue searching later.

In other words:

```suggestion
                    return i, search
        return i, None
```

That said, since currently nothing is actually using the return value, it's fine as it is for now.